### PR TITLE
MEN-4038 Hardware security subsection in Overview/Security.

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: 02.Overview/11.Security/docs.md [hunk 1]
```diff
@@ -77,3 +78,31 @@ time of one week.
...
+which opens the possibility to make use of hardware devices for cryptographic
+<!--AUTOVERSION: "www.openssl.org/docs/man%"/ignore-->
+functions via [OpenSSL Engine](https://www.openssl.org/docs/man1.1.1/man1/engine.html)
+subsystem. All you need is a proper OpenSSL configuration and an external
+shared library. You can supply the keys by PKCS#11 URI
+(see [Client configuration sections](https://docs.mender.io/client-installation/configuration-file/configuration-options#httpsclient))
+or use vendor-dependent mechanisms (see [Secure IoT with Mender and NXP EdgeLock SE050](https://hub.mender.io/t/secure-iot-with-mender-and-nxp-edgelock-se050/2744) for a more hands-on description).
+
+Currently, Mender supports hardware security engines for SSL
+handshake, mTLS, and authentication request signing.
```
Comment (mirzak) [inline 02.Overview/11.Security/docs.md:108]: Would reword this section slightly, something like this:

```
The Mender client can utilize private keys stored in [Hardware Security
Modules (HSM)](https://en.wikipedia.org/wiki/Hardware_security_module) or in 
[Trusted Platform Modules (TPM)](https://en.wikipedia.org/wiki/Trusted_Platform_Module). 
This is an additional layer of security which eliminates storage of private keys 
(secrets) as plain text files on the device, making it harder for an attacker
to gain access to keys to impersonate devices.

Starting with the Mender Client 2.4.0, the client uses OpenSSL for cryptographic
operations, which enables usage of 
[OpenSSL Engine's](https://www.openssl.org/docs/man1.1.1/man1/engine.html) as
abstractions for HSM.

For the Mender client to be able to utilize an HSM, OpenSSL must first be 
configured appropriately, and this is normally vendor specific. Please see
the following tutorial for vendor specific instructions:
- [Secure IoT with Mender and NXP EdgeLock SE050](https://hub.mender.io/t/secure-iot-with-mender-and-nxp-edgelock-se050/2744)

The Mender client supports [PKCS#11](https://en.wikipedia.org/wiki/PKCS_11), or
any other HSM access methods that are supported by 
[OpenSSL Engine's](https://www.openssl.org/docs/man1.1.1/man1/engine.html). See
[Mender client configuration sections](https://docs.mender.io/client-installation/configuration-file/configuration-options#httpsclient) for additional details.

Currently, Mender supports hardware security engines for SSL handshake, mTLS, 
and authentication request signing.
```

**Review by mirzak — CHANGES_REQUESTED** [review]

File: 02.Overview/11.Security/docs.md (hunk 1 above)
Comment (merlin-northern) [reply 02.Overview/11.Security/docs.md]: sure. I only added autoversion and changed the link to PKCS#11 to https://tools.ietf.org/html/rfc7512

**Review by mirzak — APPROVED** [review]

File: 02.Overview/11.Security/docs.md (hunk 1 above)
Comment (mirzak) [reply 02.Overview/11.Security/docs.md]: Thanks
